src/main/java/com/b4w/b4wback/repository/AuctionRepository.java

```diff
@@ -2,16 +2,52 @@
 
 import com.b4w.b4wback.dto.AuctionDTO;
 import com.b4w.b4wback.dto.CreateAuctionDTO;
+import com.b4w.b4wback.enums.GasType;
+import com.b4w.b4wback.enums.GearShiftType;
 import com.b4w.b4wback.model.Auction;
 import com.b4w.b4wback.model.User;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 import java.util.List;
 
 public interface AuctionRepository extends JpaRepository<Auction,Long> {
     List<CreateAuctionDTO> findAllByUserId(Long id);
 
     Page<AuctionDTO> findByUser(User user, Pageable pageable);
+
+
+    @Query("SELECT NEW com.b4w.b4wback.dto.AuctionDTO(auction.id, auction.title, auction.deadline, auction.status , " +
+            "COALESCE((SELECT MAX(bid.amount) FROM Bid bid WHERE bid.auction.id = auction.id), auction.basePrice)) " +
+            "FROM Auction auction WHERE " +
+            "(:milageMin IS NULL OR auction.milage >= :milageMin) AND " +
+            "(:milageMax IS NULL OR auction.milage <= :milageMax) AND " +
+            "(:modelYearMin IS NULL OR auction.modelYear >= :modelYearMin) AND " +
+            "(:modelYearMax IS NULL OR auction.modelYear <= :modelYearMax) AND " +
+            "(:brand IS NULL OR auction.brand = :brand) AND " +
```

Comment:
En el caso del brand y todos los otros campos que son strings hacer que no sean case sensitive. Puede hacerse agregandole:
`(:brand IS NULL OR lower(auction.brand) = lower(:brand))`